### ISSUE TITLE
Failed to load the atom-netconf package

### ISSUE THREAD

**Issue by venkateshld** [nokia/atom-netconf#4]
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.15.0 x64
**Electron**: 1.3.13
**OS**: Microsoft Windows 7 Enterprise 
**Thrown From**: [atom-netconf](https://github.com/nokia/atom-netconf) package 1.3.3

### Stack Trace

Failed to load the atom-netconf package

```
At Cannot find module 'atom-select-list'

Error: Cannot find module 'atom-select-list'
    at Module._resolveFilename (module.js:455:15)
    at Module._resolveFilename (~/AppData/Local/atom/app-1.15.0/resources/electron.asar/common/reset-search-paths.js:35:12)
    at Function.Module._resolveFilename (/app.asar/src/module-cache.js:383:52)
    at Function.Module._load (module.js:403:25)
    at Module.require (module.js:483:17)
    at require (/app.asar/src/native-compile-cache.js:50:27)
    at /packages/netconf/node_modules/diff/lib/diff-list-view.js:15:23)
    at Module._compile (/app.asar/src/native-compile-cache.js:109:30)
    at /app.asar/src/compile-cache.js:216:21)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
    at Function.Module._load (module.js:424:3)
    at Module.require (module.js:483:17)
    at require (/app.asar/src/native-compile-cache.js:50:27)
    at /packages/diff/lib/main.coffee:2:16)
    at /packages/diff/lib/main.coffee:1:1)
    at Module._compile (/app.asar/src/native-compile-cache.js:109:30)
    at /app.asar/src/compile-cache.js:216:21)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
    at Function.Module._load (module.js:424:3)
    at Module.require (module.js:483:17)
    at require (/app.asar/src/native-compile-cache.js:50:27)
    at /packages/netconf/lib/statusbar-netconf.coffee:15:10)
    at /packages/netconf/lib/statusbar-netconf.coffee:1:1)
    at Module._compile (/app.asar/src/native-compile-cache.js:109:30)
    at /app.asar/src/compile-cache.js:216:21)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
    at Function.Module._load (module.js:424:3)
```

### Commands

```
  2x -0:53.7.0 core:backspace (input.hidden-input)
     -0:52.9.0 core:confirm (input.hidden-input)
```

### Non-Core Packages

```
autocomplete-modules 1.6.9 
atom-netconf 1.3.3 
ssh-config 0.10.1 
```

**Comment by wisotzky** [comment]
Please provide more details. I have tried to install the NETCONF plugin freshly on my MAC using Atom 1.15 and everything works without issues. The module atom-select-list  is somewhat a replacement for atom-space-pen-views, but actually I am not using any of the two.

**Comment by cklakshmikanth** [comment]
I am facing the same issue in my Windows 10 laptop.

[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.56.0 x64
**Electron**: 9.4.4
**OS**: Microsoft Windows 10 Enterprise
**Thrown From**: [atom-netconf](https://github.com/nokia/atom-netconf) package 3.2.0


### Stack Trace

Failed to load the atom-netconf package

```
At request.charAt is not a function

TypeError: request.charAt is not a function
    at Function.Module._resolveLookupPaths (internal/modules/cjs/loader.js:632:15)
    at Module._resolveFilename (internal/modules/cjs/loader.js:780:20)
    at o._resolveFilename (electron/js2c/renderer_init.js:43:689)
    at Function.get_Module._resolveFilename (~/AppData/Local/atom/app-1.56.0/resources/app/static/<embedded>:11:157820)
    at Module.require (/app.asar/static/index.js:61:43)
    at require (~/AppData/Local/atom/app-1.56.0/resources/app/static/<embedded>:11:149153)
    at /packages/atom-netconf/lib/toolbar-netconf.coffee:13:27)
    at /packages/atom-netconf/lib/toolbar-netconf.coffee:1:1)
    at /packages/atom-netconf/lib/toolbar-netconf.coffee:1:1)
    at Module.get_Module._compile (~/AppData/Local/atom/app-1.56.0/resources/app/static/<embedded>:11:149837)
    at Object.value [as .coffee] (~/AppData/Local/atom/app-1.56.0/resources/app/static/<embedded>:11:153385)
    at Module.load (internal/modules/cjs/loader.js:815:32)
    at Module._load (internal/modules/cjs/loader.js:727:14)
    at Function.Module._load (electron/js2c/asar.js:769:28)
    at Module.require (/app.asar/static/index.js:72:46)
    at require (~/AppData/Local/atom/app-1.56.0/resources/app/static/<embedded>:11:149153)
    at /packages/atom-netconf/lib/netconf.coffee:15:18)
    at /packages/atom-netconf/lib/netconf.coffee:1:1)
    at /packages/atom-netconf/lib/netconf.coffee:1:1)
    at Module.get_Module._compile (~/AppData/Local/atom/app-1.56.0/resources/app/static/<embedded>:11:149837)
    at Object.value [as .coffee] (~/AppData/Local/atom/app-1.56.0/resources/app/static/<embedded>:11:153385)
    at Module.load (internal/modules/cjs/loader.js:815:32)
    at Module._load (internal/modules/cjs/loader.js:727:14)
    at Function.Module._load (electron/js2c/asar.js:769:28)
    at Module.require (/app.asar/static/index.js:72:46)
    at require (internal/modules/cjs/helpers.js:74:18)
    at customRequire (~/AppData/Local/atom/app-1.56.0/resources/app/static/<embedded>:1:802769)
    at requireModule (~/AppData/Local/atom/app-1.56.0/resources/app/static/<embedded>:14:1126115)
    at Package.requireMainModule (~/AppData/Local/atom/app-1.56.0/resources/app/static/<embedded>:11:3496608)
    at ~/AppData/Local/atom/app-1.56.0/resources/app/static/<embedded>:11:3483148
```

### Commands

```
```

### Non-Core Packages

```
atom-netconf 3.2.0 
```

**Comment by mkk-dev** [comment]
Doesn't work with atom-beta on Linux neither

[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.56.0 x64
**Electron**: 9.4.4
**OS**: Ubuntu 20.04.2
**Thrown From**: [atom-netconf](https://github.com/nokia/atom-netconf) package 3.2.0


### Stack Trace

Failed to load the atom-netconf package

```
At request.charAt is not a function

TypeError: request.charAt is not a function
    at Function.Module._resolveLookupPaths (internal/modules/cjs/loader.js:632:15)
    at Module._resolveFilename (internal/modules/cjs/loader.js:780:20)
    at o._resolveFilename (electron/js2c/renderer_init.js:43:689)
    at Function.get_Module._resolveFilename (/usr/share/atom/resources/app/static/<embedded>:11:157820)
    at Module.require (/app.asar/static/index.js:61:43)
    at require (/usr/share/atom/resources/app/static/<embedded>:11:149153)
    at /packages/atom-netconf/lib/toolbar-netconf.coffee:13:27)
    at /packages/atom-netconf/lib/toolbar-netconf.coffee:1:1)
    at /packages/atom-netconf/lib/toolbar-netconf.coffee:1:1)
    at Module.get_Module._compile (/usr/share/atom/resources/app/static/<embedded>:11:149837)
    at Object.value [as .coffee] (/usr/share/atom/resources/app/static/<embedded>:11:153385)
    at Module.load (internal/modules/cjs/loader.js:815:32)
    at Module._load (internal/modules/cjs/loader.js:727:14)
    at Function.Module._load (electron/js2c/asar.js:769:28)
    at Module.require (/app.asar/static/index.js:72:46)
    at require (/usr/share/atom/resources/app/static/<embedded>:11:149153)
    at /packages/atom-netconf/lib/netconf.coffee:15:18)
    at /packages/atom-netconf/lib/netconf.coffee:1:1)
    at /packages/atom-netconf/lib/netconf.coffee:1:1)
    at Module.get_Module._compile (/usr/share/atom/resources/app/static/<embedded>:11:149837)
    at Object.value [as .coffee] (/usr/share/atom/resources/app/static/<embedded>:11:153385)
    at Module.load (internal/modules/cjs/loader.js:815:32)
    at Module._load (internal/modules/cjs/loader.js:727:14)
    at Function.Module._load (electron/js2c/asar.js:769:28)
    at Module.require (/app.asar/static/index.js:72:46)
    at require (internal/modules/cjs/helpers.js:74:18)
    at customRequire (/usr/share/atom/resources/app/static/<embedded>:1:756274)
    at requireModule (/usr/share/atom/resources/app/static/<embedded>:14:1126113)
    at Package.requireMainModule (/usr/share/atom/resources/app/static/<embedded>:11:3377494)
    at /usr/share/atom/resources/app/static/<embedded>:11:3364034
```

### Commands

```
```

### Non-Core Packages

```
atom-netconf 3.2.0 
```

**Comment by wisotzky** [comment]
Interesting... The "request.charAt()" does not exist in the code of atom-netconf. Likely it is related to a framework change. I will check, what caused the issue...

**Comment by wisotzky** [comment]
@mkk-dev, could you please raise that issue against node.js please? The issue comes from there and is present since Atom 1.56. I've just talked to a colleague who run into the very same issue using Atom 1.57.
By myself I am running on MacOS X with Atom 1.55 while the issue does not exist here. For my colleague, we've uninstalled Atom 1.57 and installed 1.55 - and now everything is up and running.

In conclusion: Until this node.js issue is not resolved, you may need to go back to Atom 1.55 to get it running.

I will do a bit of investigation, what causes the issue - as other Atom modules seem to work without issues. So seems, we are hitting a code branch in node.js that others are not using.

**Comment by wisotzky** [comment]
I've raised the following issue against ATOM in the meantime:
https://github.com/atom/atom/issues/22449

**Comment by lucliu20** [comment]
@cklakshmikanth I got the same issue on my Windows 10 laptop as well.

```
Failed to load the atom-netconf package

request.charAt is not a function
Hide Stack Trace
TypeError: request.charAt is not a function
    at Function.Module._resolveLookupPaths (internal/modules/cjs/loader.js:632:15)
    at Module._resolveFilename (internal/modules/cjs/loader.js:780:20)
    at o._resolveFilename (electron/js2c/renderer_init.js:43:689)
    at Function.get_Module._resolveFilename (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:157920)
    at Module.require (file:///C:/Users/garyli/AppData/Local/atom/app-1.58.0/resources/app.asar/static/index.js:61:43)
    at require (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:149207)
    at Object.<anonymous> (file:///C:/Users/garyli/.atom/packages/atom-netconf/lib/toolbar-netconf.coffee:13:27)
    at Object.<anonymous> (file:///C:/Users/garyli/.atom/packages/atom-netconf/lib/toolbar-netconf.coffee:1:1)
    at Object.<anonymous> (file:///C:/Users/garyli/.atom/packages/atom-netconf/lib/toolbar-netconf.coffee:1:1)
    at Module.get_Module._compile (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:149891)
    at Object.value [as .coffee] (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:153485)
    at Module.load (internal/modules/cjs/loader.js:815:32)
    at Module._load (internal/modules/cjs/loader.js:727:14)
    at Function.Module._load (electron/js2c/asar.js:769:28)
    at Module.require (file:///C:/Users/garyli/AppData/Local/atom/app-1.58.0/resources/app.asar/static/index.js:72:46)
    at require (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:149207)
    at Object.<anonymous> (file:///C:/Users/garyli/.atom/packages/atom-netconf/lib/netconf.coffee:15:18)
    at Object.<anonymous> (file:///C:/Users/garyli/.atom/packages/atom-netconf/lib/netconf.coffee:1:1)
    at Object.<anonymous> (file:///C:/Users/garyli/.atom/packages/atom-netconf/lib/netconf.coffee:1:1)
    at Module.get_Module._compile (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:149891)
    at Object.value [as .coffee] (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:153485)
    at Module.load (internal/modules/cjs/loader.js:815:32)
    at Module._load (internal/modules/cjs/loader.js:727:14)
    at Function.Module._load (electron/js2c/asar.js:769:28)
    at Module.require (file:///C:/Users/garyli/AppData/Local/atom/app-1.58.0/resources/app.asar/static/index.js:72:46)
    at require (internal/modules/cjs/helpers.js:74:18)
    at customRequire (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:1:802772)
    at requireModule (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:14:1195622)
    at Package.requireMainModule (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:3520513)
    at C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:3507053
    at Package.measure (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:3505853)
    at Package.load (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:3506626)
    at PackageManager.loadAvailablePackage (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:384186)
    at PackageManager.loadPackage (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:382914)
    at PackageManager.activatePackage (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:385486)
    at C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:915033
    at r (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:909639)
    at n (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:14:1194945)
    at ChildProcess.<anonymous> (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:14:1195151)
    at ChildProcess.emit (events.js:223:5)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
The error was thrown from the atom-netconf package. This issue has already been reported.
```

ATOM Version 1.58.0 x64

**Comment by selvaitech** [comment]
> @cklakshmikanth I got the same issue on my Windows 10 laptop as well.
> 
> ```
> Failed to load the atom-netconf package
> 
> request.charAt is not a function
> Hide Stack Trace
> TypeError: request.charAt is not a function
>     at Function.Module._resolveLookupPaths (internal/modules/cjs/loader.js:632:15)
>     at Module._resolveFilename (internal/modules/cjs/loader.js:780:20)
>     at o._resolveFilename (electron/js2c/renderer_init.js:43:689)
>     at Function.get_Module._resolveFilename (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:157920)
>     at Module.require (file:///C:/Users/garyli/AppData/Local/atom/app-1.58.0/resources/app.asar/static/index.js:61:43)
>     at require (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:149207)
>     at Object.<anonymous> (file:///C:/Users/garyli/.atom/packages/atom-netconf/lib/toolbar-netconf.coffee:13:27)
>     at Object.<anonymous> (file:///C:/Users/garyli/.atom/packages/atom-netconf/lib/toolbar-netconf.coffee:1:1)
>     at Object.<anonymous> (file:///C:/Users/garyli/.atom/packages/atom-netconf/lib/toolbar-netconf.coffee:1:1)
>     at Module.get_Module._compile (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:149891)
>     at Object.value [as .coffee] (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:153485)
>     at Module.load (internal/modules/cjs/loader.js:815:32)
>     at Module._load (internal/modules/cjs/loader.js:727:14)
>     at Function.Module._load (electron/js2c/asar.js:769:28)
>     at Module.require (file:///C:/Users/garyli/AppData/Local/atom/app-1.58.0/resources/app.asar/static/index.js:72:46)
>     at require (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:149207)
>     at Object.<anonymous> (file:///C:/Users/garyli/.atom/packages/atom-netconf/lib/netconf.coffee:15:18)
>     at Object.<anonymous> (file:///C:/Users/garyli/.atom/packages/atom-netconf/lib/netconf.coffee:1:1)
>     at Object.<anonymous> (file:///C:/Users/garyli/.atom/packages/atom-netconf/lib/netconf.coffee:1:1)
>     at Module.get_Module._compile (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:149891)
>     at Object.value [as .coffee] (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:153485)
>     at Module.load (internal/modules/cjs/loader.js:815:32)
>     at Module._load (internal/modules/cjs/loader.js:727:14)
>     at Function.Module._load (electron/js2c/asar.js:769:28)
>     at Module.require (file:///C:/Users/garyli/AppData/Local/atom/app-1.58.0/resources/app.asar/static/index.js:72:46)
>     at require (internal/modules/cjs/helpers.js:74:18)
>     at customRequire (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:1:802772)
>     at requireModule (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:14:1195622)
>     at Package.requireMainModule (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:3520513)
>     at C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:3507053
>     at Package.measure (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:3505853)
>     at Package.load (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:3506626)
>     at PackageManager.loadAvailablePackage (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:384186)
>     at PackageManager.loadPackage (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:382914)
>     at PackageManager.activatePackage (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:385486)
>     at C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:915033
>     at r (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:11:909639)
>     at n (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:14:1194945)
>     at ChildProcess.<anonymous> (C:\Users\garyli\AppData\Local\atom\app-1.58.0\resources\app\static\<embedded>:14:1195151)
>     at ChildProcess.emit (events.js:223:5)
>     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
> The error was thrown from the atom-netconf package. This issue has already been reported.
> ```
> 
> ATOM Version 1.58.0 x64

Same issue i am facing in my Windows 10 Laptop as well as 18.04 Ubuntu. Can someone help us here?

**Comment by wisotzky** [comment]
The `request.charAt` issue should be fixed now with version 3.3.1.
Could you please validate, if it now works in your environments?

**Comment by wisotzky** [comment]
Fixed by:
https://github.com/nokia/atom-netconf/commit/53bd8c3fec2cd3dd0a01c3480cd6a8d157a012dc#diff-76b5a357391276b282a516f54f48ef3c207f46d8192dc58c208d5183d38415f8

**Comment by lucliu20** [comment]
![image](https://user-images.githubusercontent.com/76449030/128097140-dd1198b8-9726-40fd-990f-20fd4251a3f1.png)
Upgraded from version 3.3.0 to version 3.3.1 on Windows 10 Thinkpad laptop. The Settings button is not available. Nowhere to view and change the Netconf server info. Also clicked on the Uninstall button to remove it, but no luck.

Never mind. Uninstalled Atom completed, then re-downloaded Atom 1.58.0 x64, installed it, and then installed version 3.3.1. It works fine. Thanks.

**Comment by wisotzky** [comment]
@lucliu20, great to hear that it now works for you. Please be aware, that the settings page is the regular settings page of the plugin - so the taskbar icon is just a shortcut for this. Also be aware, that it is possible to provide `servers.yaml` file. While it is not possible to change this file through the `settings` dialogue, it allows you to have multiple servers defined to connect to (only one at a time) and to share this file among multiple clients.

**Comment by lucliu20** [comment]
> @lucliu20, great to hear that it now works for you. Please be aware, that the settings page is the regular settings page of the plugin - so the taskbar icon is just a shortcut for this. Also be aware, that it is possible to provide `servers.yaml` file. While it is not possible to change this file through the `settings` dialogue, it allows you to have multiple servers defined to connect to (only one at a time) and to share this file among multiple clients.

The multiple servers feature comes in handy. Thanks!

![2021-08-17 10_52_00-](https://user-images.githubusercontent.com/76449030/129776542-5c2e63d4-d454-4103-a58a-8c812a2490c5.png)
